content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,281 @@
+server.feeStats().then(function (response) {
+  let avgFee = parseFloat(response.fee_charged.p50);
+  let tx = base.TransactionBuilder(someAccount, {
+    fee: (avgFee * 1.10).toFixed(0),  // bump & convert to int
+    networkPassphrase: // ...
+  });
+  // ...build the rest of the tx... 
+  tx.sign(someAccount);
+  return server.submitTransaction(tx);
+});
+```
+
+</CodeExample>
+
+##### Bumping fees on past transactions
+It's possible that even with the liberal fee-paying policy, your transaction fails to make it into the ledger due to insufficient funds or untimely surges. Resolving this problem is exactly the goal of a [fee-bump transaction](https://developers.stellar.org/docs/glossary/fee-bumps/); the following snippet shows how you can resubmit a transaction with a higher fee given that you have the original [transaction envelope](https://developers.stellar.org/docs/glossary/transactions/#transaction-envelopes):
+
+<CodeExample>
+
+```js
+// Let `lastTx` be some transaction that fails submission due to high fees, and
+// `lastFee` be the maximum fee (expressed as an int) willing to be paid by 
+// `account` for `lastTx`.
+server.submitTransaction(lastTx).catch(function (error) {
+  if (isFeeError(error)) {
+    let bump = sdk.TransactionBuilder.buildFeeBumpTransaction(
+      account,      // account that will PAY the new fee
+      lastFee * 10, // new fee
+      lastTx,
+      server.networkPassphrase
+    );
+    bump.sign(someAccount);
+    return server.submitTransaction(bump);
+  }
+  // ...other error conditions...
+}).then(...);
+```
+
+</CodeExample>
+
+Note an [important stipulation](https://developers.stellar.org/docs/glossary/fee-bumps/#replace-by-fee) of fee bumping that's fulfilled above:
+
+> If you submit two distinct transactions with the same source account and sequence number, and the second transaction is a fee-bump transaction, the second transaction will be included in the transaction queue in place of the first transaction if and only if **the fee bid of the second transaction is at least 10x the fee bid of the first transaction**.
+
+
+### Timeouts
+If you receive a 504 error from Horizon after a transaction submission, you've encountered a timeout. In the Stellar universe, a timeout indicates something a little nebulous and non-traditional: it's more of a *warning* that your request hasn't been fulfilled within a reasonable amount of time *yet* more than an error. This subtlety arises because of the nature of the relationship between Horizon and Stellar Core: the network might take some time to accept a transaction---especially one with a low fee, see [Surge Pricing](#surge-pricing) later---on the order of 5-10 minutes in the worst cases, whereas Horizon needs to provide developers with a response within 30 seconds or so.
+
+This leads to a very important point: **receiving a 504 for your transaction submission does not mean the transaction did not make it to the network**.
```

Comment:
TIL x2! Great info here 💯 

question @Shaptic , if you don’t set timebounds on a transaction what’s the upperbound on the time Core will try to add your transaction to the ledger? (eg. if theoretically there are always competing transactions with higher baseFees during surge pricing, when does Core give up on your transaction)

I see you put `5-10 minutes` is there a hard max value for that? Either way I think adding a link or short note on Core's max effort for your transaction would be a helpful note here

Reply:
I think you got your answer in Slack @acharb, but for posterity, it appears that transactions w/o timebounds will **typically** be considered for 3 ledgers then banned for 10, but this can go on forever.